commands_to_load/Here.py

```diff
@@ -28,26 +28,24 @@ def build_embed(members, channel):
         elif len(members) > 50:
             string = "There's a lot of people here.\n"
         else:
-            string =  "The following users have permission for this channel.\n"
-            nicks = ""
-            names = ""
-            for member in members:
-                nicks += member.display_name + "\n"
-                names += str(member) + "\n"
+            string =  "The following (" + str(len(members)) + ") users have permission for this channel.\n"
+
+            # newline separated lists of members and their nicknames
+            nicks = "\n".join([member.display_name for member in members])
+            names = "\n".join([str(member) for member in members])
 
             embed.add_field(name = "Name", value = nicks, inline = True)
             embed.add_field(name = "Account", value = names, inline = True)
 
-        roles = ""
-        for role in channel.changed_roles:
-            if roles == "":
-                roles += "@" + role.name
-            else:
-                roles += ", @" + role.name
-        if roles == "":
-            roles = "@everyone"
+        # comma separated list of role names for each role in the channel
+        # if they can read messages
+        # like how it says...
+        roles = ", ".join([role.name
```

Comment:
whats the benefit of this "channel specific role" section if it can only work with roles that have channel specific read access and not also include inherited read access?

Reply:
what do you mean by inherited read access

Reply:
like how execs have read access to every channel by default, they dont need have channel specific ready access to every channel. [if i understand discord permissions correctly]

Reply:
first of all: execs don't have read access to every channel.
second, if a role did have read access to every channel, listing them would be redundant because you know they have access to every channel already.

Reply:
The idea was to have a list of roles you could assign someone if you were looking to add them to the chat, and also know what groups of people the channel is intended for.

Reply:
>first of all: execs don't have read access to every channel.

I was not aware of that.

and OK, your reasons make sense.